src/beacon/drand.rs

```diff
@@ -260,31 +240,68 @@ impl DrandBeacon {
 
 #[async_trait]
 impl Beacon for DrandBeacon {
-    fn verify_entry(&self, curr: &BeaconEntry, prev: &BeaconEntry) -> Result<bool, anyhow::Error> {
-        // TODO(forest): https://github.com/ChainSafe/forest/issues/3572
-        if prev.round() == 0 {
-            return Ok(true);
-        }
+    fn verify_entries<'a>(
+        &self,
+        entries: &'a [BeaconEntry],
+        mut prev: &'a BeaconEntry,
+    ) -> Result<bool, anyhow::Error> {
+        if self.network.is_unchained() {
+            let mut messages = vec![];
+            let mut signatures = vec![];
+            let pk = PublicKeyOnG2::from_bytes(&self.public_key)?;
+            for entry in entries.iter() {
+                // Hash the messages (H(curr_round))
+                let message = {
+                    let mut round_bytes = [0; std::mem::size_of::<u64>()];
+                    BigEndian::write_u64(&mut round_bytes, entry.round());
+                    sha2::Sha256::digest(round_bytes)
+                };
+                messages.push(message);
+                signatures.push(SignatureOnG1::from_bytes(entry.signature())?);
+            }
 
-        // Hash the messages (H(prev sig | curr_round))
-        let digest = {
-            let mut round_bytes = [0; std::mem::size_of::<u64>()];
-            BigEndian::write_u64(&mut round_bytes, curr.round());
-            let mut hasher = sha2::Sha256::default();
-            hasher.update(prev.data());
-            hasher.update(round_bytes);
-            hasher.finalize()
-        };
-        // Signature
-        let sig = Signature::from_bytes(curr.data())?;
-        let sig_match = bls_signatures::verify_messages(&sig, &[&digest], &[self.pub_key.key()?]);
```

Comment:
`bls_signatures::verify_messages` is designed for validating `bls_messages` in full tipsets, with 1 signature and a list of (message, public key) pairs to validate. While for drand beacon verification, there are 1 public key and a list of (message, signature) pairs to validate, so re-writing the logic to allow benefiting from parallelization.